src/functions/text.js

```diff
@@ -61,9 +61,11 @@ defineFunction({
     },
     htmlBuilder(group, options) {
         const newOptions = optionsWithFont(group, options);
+        // TODO(kevinb): split group.body into combinable parts to avoid having
+        // complicated checks in buildCommon.tryCombineChars.
```

Comment:
The only things we can combine are `textord`s and `spacing`s so if we create groups of consecutive `textord`s and `spacing`s we can pass each of those groups off to `buildExpression` and combine the `symbolNodes` that are returned without doing all of complex checks we're doing right now in `tryCombineChars`.

Reply:
I worry that some other functions might return the equivalent of a `textord` that wouldn't be caught before running the function...